app/main.py

```
@@ -1,34 +1,97 @@
+        for deck in self.decks:
+            if (row, column) == deck.cell:
+                deck.is_alive = False
+        self.is_drowned = not any([deck.is_alive for deck in self.decks])
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+
+    def __init__(self, ships: list) -> None:
+        self.field = {}
+        for ship in ships:
+            ship = Ship(ship[0], ship[1])
+            for deck in ship.decks:
+                self.field[deck.cell] = ship
+
+        self._validate_field()
+
+    def fire(self, location: tuple) -> str:
+        if location in self.field:
+            self.field[location].fire(location[0], location[1])
+            if self.field[location].is_drowned:
+                return "Sunk!"
+            return "Hit!"
+        return "Miss!"
+
+    def _validate_field(self) -> None:
+        len_is_10 = len(self.field) == 20
+        all_ships = []
+        for ship in set(self.field.values()):
+            all_ships.append(len(ship.decks))
+
+        one_four_deck = all_ships.count(4) == 1
+        two_triple_deck = all_ships.count(3) == 2
+        three_double_deck = all_ships.count(2) == 3
+        four_single_deck = all_ships.count(1) == 4
+
+        if all((
+                len_is_10,
+                one_four_deck,
+                two_triple_deck,
+                three_double_deck,
+                four_single_deck
+        )):
+            print("Field is validated")
+        else:
+            print("You break the rules of battleship game")
```

Better to raise an Exception is something went wrong